sgqlc/endpoint/requests.py

```diff
@@ -109,7 +109,13 @@ def __init__(self, url, base_headers=None, timeout=None, method='POST',
         self.timeout = timeout
         self.method = method
         self.auth = auth
-        self.session = session
+        self.session = session or requests.Session()
+        self.close_session = session is not None
+
+    def __del__(self):
+        # keep the session open if it has been provided by the user
+        if self.session and self.close_session:
```

Comment:
the first `if self.session` will always be true

Reply:
I've added it so there's no case in the future where the session may be `None` and an exception is thrown here.